src/axom/inlet/examples/nested_structs.cpp

```diff
@@ -0,0 +1,276 @@
+    {
+      result.type = Operator::Type::Rotate;
+      result.rotate = base["rotate"];
+      result.axis = mapToVector(base["axis"]);
+      result.center = mapToVector(base["center"]);
+    }
+    else if(base.contains("slice"))
+    {
+      result.type = Operator::Type::Slice;
+      // Grab the substructure corresponding to the slice operation
+      // and use it to populate the Operator instance
+      auto slice = base["slice"];
+      if(slice.contains("x"))
+      {
+        result.x = slice["x"];
+      }
+      if(slice.contains("y"))
+      {
+        result.y = slice["y"];
+      }
+      if(slice.contains("z"))
+      {
+        result.z = slice["z"];
+      }
+    }
+    return result;
+  }
+};
+
+std::ostream& operator<<(std::ostream& os, const Operator& op)
+{
+  switch(op.type)
+  {
+  case Operator::Type::Translate:
+    os << "   Translation operator:\n";
+    os << fmt::format("      with vector: {0}\n", op.translate);
+    break;
+  case Operator::Type::Rotate:
+    os << "   Rotation operator:\n";
+    os << fmt::format("      with axis: {0}\n", op.axis);
+    os << fmt::format("      with center: {0}\n", op.center);
+    break;
+  case Operator::Type::Slice:
+    os << "   Slice operator:\n";
+    os << fmt::format("      with x-coord: {0}\n", op.x);
+    os << fmt::format("      with y-coord: {0}\n", op.y);
+    os << fmt::format("      with z-coord: {0}\n", op.z);
+    break;
+  }
+  return os;
+}
+
+struct Geometry
+{
+  std::vector<Operator> operators;
+  std::string format;
+  std::string path;
+  enum class Units
+  {
+    Centimeters,
+    Meters
+  } units;
+  int start_dim;
+  static void defineSchema(inlet::Table& table)
+  {
+    table.addString("format");
+    table.addString("path");
+    // A string is used to represent the enumeration
+    table.addString("units");
+    table.addInt("start_dimensions");
+    // addGenericArray is used to represent std::vector<T> where
+    // T is any non-primitive type
+    auto& ops_table = table.addGenericArray("operators");
+    Operator::defineSchema(ops_table);
+  }
+};
+
+template <>
+struct FromInlet<Geometry>
+{
+  Geometry operator()(const inlet::Table& base)
+  {
+    Geometry result;
+    result.format = base["format"];
+    result.path = base["path"];
+    result.start_dim = base["start_dimensions"];
+
+    // FIXME: Remove when PR #429 is merged
+    auto ops = base["operators"].get<std::unordered_map<int, Operator>>();
+    for(const auto& ele : ops)
+    {
+      result.operators.push_back(ele.second);
+    }
+    return result;
+  }
+};
+
+std::ostream& operator<<(std::ostream& os, const Geometry& geom)
+{
+  os << fmt::format("Geometry in format: '{0}'\n", geom.format);
+  os << fmt::format("  with path: '{0}'\n", geom.path);
+  for(const auto& op : geom.operators)
+  {
+    os << op;
+  }
+  return os;
+}
+
+struct Shape
+{
+  std::string name;
+  enum class Material
+  {
+    Steel,
+    Wood,
+    Plastic
+  } material;
```

Comment:
Does the `Material` enum and/or `material` instance add anything essential to this example? e.g. w.r.t. supporting this use case?

Tag: @estebanpauli

Reply:
FWIW, in Klee, this is string, not an enum. If this serves to verify that enums work, though, or shows how to use them, I think it helps the example.

Reply:
In this example enumerations are represented in the input deck as strings, so it would be easy use that approach.

I agree that the enum should be left in, as it may be instructive for future users looking to "deserialize" enumerations with Inlet.